src/ophys_etl/modules/segmentation/postprocess_utils/roi_time_correlation.py

```diff
@@ -0,0 +1,204 @@
+    th = np.quantile(key_pixel, discard)
+    mask = (key_pixel >= th)
+    sub_video = sub_video[mask, :]
+    key_pixel = key_pixel[mask]
+    ntime = len(key_pixel)
+    assert key_pixel.shape == (ntime,)
+    assert sub_video.shape == (ntime, npix)
+
+    key_mu = np.mean(key_pixel)
+    vid_mu = np.mean(sub_video, axis=0)
+    assert vid_mu.shape == (npix,)
+    key_var = np.mean((key_pixel-key_mu)**2)
+    sub_vid_minus_mu = sub_video-vid_mu
+    sub_var = np.mean(sub_vid_minus_mu**2, axis=0)
+    assert sub_var.shape == (npix,)
+
+    numerator = np.dot((key_pixel-key_mu), sub_vid_minus_mu)/ntime
+    assert numerator.shape == (npix,)
+
+    corr = numerator/np.sqrt(sub_var*key_var)
+    assert corr.shape == (npix,)
+    return corr
+
+
+def get_brightest_pixel(roi: SegmentationROI,
+                        img_data: np.ndarray,
+                        video_data: np.ndarray) -> np.ndarray:
+    """
+    Return the brightest pixel in an ROI (as measured against
+    some image) as a time series.
+
+    Parameters
+    ----------
+    roi: SegmentationROI
+
+    img_data: np.ndarray
+        The image used to assess "brightest pixel".
+        Shape is (nrows, ncols).
+
+    video_data: np.ndarray
+        Shape is (ntime, nrows, ncols)
+
+    Returns
+    -------
+    brightest_pixel: np.ndarray
+        Time series of taken from the video at the
+        brightest pixel in the ROI
+    """
+    xmin = roi.x0
+    ymin = roi.y0
+    xmax = roi.x0+roi.width
+    ymax = roi.y0+roi.height
+    sub_img = img_data[ymin:ymax, xmin:xmax]
+    sub_video = video_data[:, ymin:ymax, xmin:xmax]
+    mask = roi.mask_matrix
+    sub_img = sub_img[mask]
+    sub_video = sub_video[:, mask]
+    assert len(sub_video.shape) == 2
+    assert sub_video.shape[1] == sub_img.shape[0]
+    assert len(sub_img.shape) == 1
+    brightest_pixel = np.argmax(sub_img)
+    return sub_video[:, brightest_pixel]
+
+
+def validate_merger_corr(uphill_roi: SegmentationROI,
+                         downhill_roi: SegmentationROI,
+                         video_data: np.ndarray,
+                         img_data: np.ndarray,
+                         filter_fraction: float = 0.2,
+                         acceptance: float = 1.0) -> bool:
+    """
+    Validate the merger between two ROIs based on time correlation
+    information.
+
+    Parameters
+    ----------
+    uphill_roi: SegmentationROI
+
+    downhill_roi: SegmentationROI
+
+    video_data: np.ndarray
+        Shape is (ntime, nrows, ncols)
+
+    img_data: np.ndarray
+        Shape is (nrows, ncols)
+
+    filter_fraction: float
+        The fraction of time steps to keep when doing
+        time correlation (default = 0.2)
+
+    acceptance: float
+        The z-score threshold for accepting the merger
+        (default = 1.0)
+
+    Returns
+    -------
+    boolean
+
+    Notes
+    -----
+    To assess whether a merger should happen, find the
+    brightest pixel in uphill_roi (reckoned with img_data).
```

Comment:
what is the importance of brightest pixel?
It seems like this check could be made regardless of which pixel.
For example, you could average all the ROI pixel traces and make your correlations of individual pixels against that ROI average.
It seems wonky to be making judgments about correlations, which are based on, and presumably sensitive to, some image input that may or may not be related to correlation.